docs/user_v2/how-to/make-a-device.rst

```diff
@@ -1,4 +1,84 @@
+.. note::
+
+    Ophyd.v2 is included on a provisional basis until the v2.0 release and 
+    may change API on minor release numbers before then
+
 Make a Device
 =============
 
-TODO
\ No newline at end of file
+.. currentmodule:: ophyd.v2.core
+
+To make a Device using Ophyd.v2, you need to subclass from the `Device`
+class, create some `Signal` instances, implement :meth:`~Device.connect()` and
+implement some other suitable Bluesky `hardware_interface`.
+
+The rest of this guide will show examples from ``ophyd/v2/epicsdemo/__init__.py``
+
+Readable
+--------
+
+For a simple `bluesky.protocols.Readable` object like a `Sensor`, you could just
+implement these methods directly:
+
+.. literalinclude:: ../../../ophyd/v2/epicsdemo/__init__.py
+   :pyobject: Sensor
+
+The PV prefix is passed to the constructor and stored, then Signals constructed
+and stored on the Device. Each one is passed its Python type, which could be a
+primitive (`str`, `int`, `float`), an array (`numpy.typing.NDArray`), or an enum
+(`enum.Enum`). It is also passed the PV suffix. Finally ``set_name()`` is called
+which will give a name to the Device and all its child Devices.
+
+The ``connect()`` method takes an additional prefix (in case this Device is nested
+in another) and passes them to the `connect_children()` function that will
+connect all the Signals in parallel.
+
+All the other methods directly implement the Bluesky protocols by calling
+down to a single relevant Signal. This means that only ``energy`` will be
+reported as configuration, and only ``value`` will be reported in read.
+
+Movable
+-------
+
+For a more complicated `bluesky.protocols.Readable` device like a `Mover`, you
+can use the `HasReadableSignals` mix-in class:
+
+.. literalinclude:: ../../../ophyd/v2/epicsdemo/__init__.py
+   :pyobject: Mover
```

Comment:
I notice that for these v2 Devices, the PV structures are not known until runtime at `__init__`. Is there any way to inspect the device structures from the class as for a v1 Device? I realize that connection is no longer automatically done in `__init__` so this might not matter as much as it does in v1.

Reply:
There isn't at runtime, but `mypy` knows about it, so can provide static analysis of your code. It also knows about the types of `signal.get_value()` which means it can check complicated flyer code which might put and get many signals.

Is there a usecase for introspecting which signals a class would create before it is created?

Reply:
ophyd v1 exposes some classmethods that I've found useful on occasion e.g. https://github.com/bluesky/ophyd/blob/0e8acb3df3d17e0ab7aa2cc924831f4a0c580449/ophyd/device.py#L985
but I can't immediately come up with specific reason that they'd need to be run before creation.

I think I instinctively want these things because instantiating an ophyd v1 device can be slow. So if that's no longer the case then we're in a good spot.

Reply:
I think ophyd v1 was weird to me at first too, I don't think there's a specific part of the chain here that is lacking or a step backwards. I think this is just a familiarity thing.